pulsar-zookeeper-utils/src/main/java/org/apache/pulsar/zookeeper/LocalBookkeeperEnsemble.java

```diff
@@ -222,9 +222,17 @@ private void runBookies(ServerConfiguration baseConf) throws Exception {
                 cleanDirectory(bkDataDir);
             }
 
+            int bookiePort = initialPort + i;
+
+            // Ensure registration Z-nodes are cleared when standalone service is restarted ungracefully
+            String registrationZnode = String.format("/ledgers/available/%s:%d", baseConf.getAdvertisedAddress(), bookiePort);
+            if (zkc.exists(registrationZnode, null) != null) {
+                zkc.delete(registrationZnode, -1);
```

Comment:
handle NoNode exception?

Reply:
You mean to handle the case where the node is expired between exists() and delete()? Sure, I'll add that.